modules/tinymce/src/core/test/ts/browser/newline/InsertNewLineTest.ts

```diff
@@ -836,4 +836,15 @@ describe('browser.tinymce.core.newline.InsertNewLineTest', () => {
     TinyAssertions.assertContent(editor, '<blockquote><p>A</p></blockquote><p>&nbsp;</p>');
     TinyAssertions.assertCursor(editor, [ 1, 0, 0, 0 ], 0);
   });
+
+  it('TINY-10560: Press Enter after a `selection.setContent` should create a new line and not throw errors', () => {
```

Comment:
Could you stick this into a test function then feed it more scenarios. H1 etc seems that more blocks can produce the error.

You could also simply set the selection to the spot where this happens the insert code is really not part of the bug the bug is that if you press enter if the selection range is after the text node then it blows up.

You can still have one specific test case of the pre and insert so that we cover the scenario described in the ticket.

So this should also exhibit the bug:
```
editor.setContent('<p>abc</p>');
TinySelections.setCursor(editor, [ 0 ], 1);
insertNewline(editor, { });
```

Reply:
I refactored the test, now I do the check for different tags and removed the content expectations, I didn't change the selection to put the caret, let me know if it is ok in this way or if I should change something else